test/albumController.test.js

```diff
@@ -0,0 +1,190 @@
+    const albumToCreate = {
+      albumName: 'Album 1',
+      artistName: 'Artist 1',
+      spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+    };
+
+    await Album.create(albumToCreate);
+
+    const response = await request(app).get(
+      '/api/v1/albums?limit=10&order=asc'
+    );
+
+    expect(response.status).toBe(StatusCodes.OK);
+    // Assert that the response body is an object with "albums" and "count" keys
+    expect(response.body).toHaveProperty('albums');
+    expect(response.body).toHaveProperty('count');
+    // Assert that "albums" is an array with the expected number of albums (based on the limit)
+    // In this case, since we have only 1 album in the database, the "albums" array should have 1 item
+    expect(response.body.albums).toHaveLength(1);
+
+    // Assert that the "count" field reflects the actual count of albums in the database
+    // In this case, it should be 1 because we created 1 album
+    expect(response.body.count).toBe(1);
+  });
+
+  it('should return a limited list of albums when using a limit parameter', async () => {
+    const albumsToCreate = [
+      {
+        albumName: 'Album 1',
+        artistName: 'Artist 1',
+        price: 9.99,
+        image: '/uploads/album1.jpg',
+        releaseDate: '2022-01-01T00:00:00.000Z',
+        category: 'rock',
+        spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+        averageRating: 4.5,
+        numOfReviews: 2,
+      },
+      {
+        albumName: 'Album 2',
+        artistName: 'Artist 2',
+        price: 8.99,
+        image: '/uploads/album2.jpg',
+        releaseDate: '2022-02-01T00:00:00.000Z',
+        category: 'pop',
+        spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+        averageRating: 3.7,
+        numOfReviews: 3,
+      },
+      {
+        albumName: 'Album 3',
+        artistName: 'Artist 3',
+        price: 7.99,
+        image: '/uploads/album3.jpg',
+        releaseDate: '2022-03-01T00:00:00.000Z',
+        category: 'rock',
+        spotifyUrl: 'https://api.spotify.com/v1/albums/blah123',
+        averageRating: 4.2,
+        numOfReviews: 1,
+      },
+    ];
+
+    await Album.insertMany(albumsToCreate);
+
+    // Make a GET request to the /api/v1/albums endpoint with a limit parameter set to 2
+    const limit = 2;
+    const response = await request(app).get(`/api/v1/albums?limit=${limit}`);
+
+    // Assert that the response status is OK (200)
+    expect(response.status).toBe(StatusCodes.OK);
+
+    // Assert that the response body is an object with "albums" and "count" keys
+    expect(response.body).toHaveProperty('albums');
+    expect(response.body).toHaveProperty('count');
+
+    // Assert that "albums" is an array with a length equal to the specified limit
+    expect(response.body.albums).toHaveLength(limit);
+
+    // Assert that the "count" field reflects the actual count of albums in the database
+    // In this case, it should be 5 because we created 5 albums
```

Comment:
We created 3 albums here and there's also the album that gets created before each test in the `beforeEach`. So I would expect that there are 4 total albums in the db.

However, this endpoint returns `count` property that's just the count of albums _returned_ in this response: https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/ed2a4e7e04bf48fce393395a8e00a4f3975c6549/src/controllers/albumController.js#L188

Reply:
I have changed the limit of albums for 3 and test results successful, it seems it calculates these albums which are in block